engine/class_modules/monk/sc_monk.cpp

```diff
@@ -4306,6 +4480,186 @@ struct yulon_spell_t : public monk_spell_t
+    // Expiring this before the execute so that there is an easy way to apply the damage bonus.
+    p()->buff.unity_within->expire();
+
+    monk_spell_t::execute();
+  }
+};
+
+// ==========================================================================
+// Celestial Conduit
+// ==========================================================================
+
+// This is a separate spell from the normal Flight of the Red Crane damage spell
+struct flight_of_the_red_crane_celestial_dmg_t : public monk_spell_t
+{
+  flight_of_the_red_crane_celestial_dmg_t( monk_t *p )
+    : monk_spell_t( p, "flight_of_the_red_crane_celestial_dmg",
+                    p->talent.conduit_of_the_celestials.flight_of_the_red_crane_celestial_dmg )
+  {
+    background = true;
+    aoe        = p->talent.conduit_of_the_celestials.flight_of_the_red_crane->effectN( 1 ).base_value();
+  }
+
+  double composite_aoe_multiplier( const action_state_t *state ) const override
+  {
+    double cam = monk_spell_t::composite_aoe_multiplier( state );
+
+    if ( p()->buff.unity_within->check() )
+      cam *= 1 + p()->talent.conduit_of_the_celestials.unity_within_dmg_mult->effectN( 1 ).percent();
+
+    return cam;
+  }
+
+  void execute() override
+  {
+    monk_spell_t::execute();
+
+    p()->buff.flight_of_the_red_crane->trigger();
+  }
+};
+
+struct flight_of_the_red_crane_celestial_heal_t : public monk_heal_t
+{
+  flight_of_the_red_crane_celestial_heal_t( monk_t *p )
+    : monk_heal_t( p, "flight_of_the_red_crane_celestial",
+                   p->talent.conduit_of_the_celestials.flight_of_the_red_crane_celestial_dmg )
+  {
+    background = true;
+    aoe        = p->talent.conduit_of_the_celestials.flight_of_the_red_crane->effectN( 1 ).base_value();
+    target     = p;
+  }
+
+  double composite_aoe_multiplier(const action_state_t* state) const override
+  {
+    double cam = monk_heal_t::composite_aoe_multiplier( state );
+
+    if ( p()->buff.unity_within->check() )
+      cam *= 1 + p()->talent.conduit_of_the_celestials.unity_within_dmg_mult->effectN( 1 ).percent();
+
+    return cam;
+  }
+};
+
+struct celestial_conduit_t : public monk_spell_t
+{
+  struct celestial_conduit_dmg_t : public monk_spell_t
+  {
+    celestial_conduit_dmg_t( monk_t *p )
+      : monk_spell_t( p, "celestial_conduit_dmg", p->talent.conduit_of_the_celestials.celestial_conduit_dmg )
+    {
+      background = true;
+    }
+
+    double composite_aoe_multiplier( const action_state_t *state ) const override
+    {
+      double cam = monk_spell_t::composite_aoe_multiplier( state );
+
+      const std::vector<player_t *> &targets = state->action->target_list();
+
+      if ( !targets.empty() )
+        cam *= 1 + ( p()->talent.conduit_of_the_celestials.celestial_conduit->effectN( 1 ).percent() *
+                     std::min( (double)targets.size(),
+                               p()->talent.conduit_of_the_celestials.celestial_conduit->effectN( 3 ).base_value() ) );
+
+      return cam;
+    }
```

Comment:
`state->n_targets` is the number of targets the entire execution hits.

There may be an adaptable set of action settings, as used by bfa-style trinket scaling. May not be able to override the appropriate constants, though.